src/theme/typography/helper.ts

```diff
@@ -0,0 +1,99 @@
+import { css } from "styled-components";
+import { StyledComponentProps, getCollection, getValue } from "../helpers";
+import { FontScheme, ThemeCollectionSpec } from "../types";
+import { LifeSgTypographySet } from "./specs/typography-set";
+import { TypoGraphyCollectionMap, TypographySet } from "./types";
+
+const TypographySpec: ThemeCollectionSpec<TypoGraphyCollectionMap, FontScheme> =
+    {
+        collections: {
+            lifesg: LifeSgTypographySet,
+            bookingsg: LifeSgTypographySet,
+            rbs: LifeSgTypographySet,
+            mylegacy: LifeSgTypographySet,
+            ccube: LifeSgTypographySet,
+        },
+        defaultValue: "lifesg",
+    };
+
+export const getTypography = (key: keyof TypographySet) => {
+    return (props: StyledComponentProps) => {
+        const theme = props.theme;
+        const typographySet: TypographySet = getCollection(
+            TypographySpec,
+            theme["fontScheme"]
+        );
+
+        // Check for an override
+        const typographyValue =
+            theme.overrides && theme.overrides.typography
+                ? getValue(typographySet, key, theme.overrides.typography)
+                : typographySet[key];
+
+        // If function, resolve with props
+        if (typeof typographyValue === "function") {
+            return css`
+                ${(typographyValue as (props: any) => string)(props)}
+            `;
+        }
+
+        return css`
+            ${typographyValue}
+        `;
+    };
+};
+
+export const TypographyValues = {
```

Comment:
type it to narrow the keys and values, do the same for `FontValues`

```suggestion
export const TypographyValues: { [key in keyof TypographySet]: (props: StyledComponentProps) => CSSProp } = {
```